### ISSUE TITLE
Deleting defer and only queries error

### ISSUE THREAD

**Issue by cfytrok** [jazzband/django-simple-history#678]
Deleting objects that were quered using only or defer results in DoesNotExist error.

```
class A(models.Model):
    name = models.CharField(max_length=255)
    history = HistoricalRecords()

A.objects.create(name='a')
objs = A.objects.only('pk').filter(id=1).all()
objs.delete()
```

Error raised in create_historical_record method, when django tries to get other fields from db, while object doesn't exist any more.

Possible solution: query all fields for deleted objects on pre_delete signal

**Environment (please complete the following information):**
 - Django Simple History Version: 2.10.0
 - Django Version: 3.0.3

**Comment by JordanHyatt** [comment]
Just submitted pull request #1064 to address this

**Comment by ddabble** [comment]
Resolved by #1393
